src/internal/item-container/index.tsx

```diff
@@ -98,7 +99,7 @@ export interface ItemContainerProps {
     maxHeight: number;
   };
   onKeyMove?(direction: Direction): void;
-  children: () => ReactNode;
+  children: (showPreview?: boolean) => ReactNode;
```

Comment:
This line is the problem. `showPreview` only makes sense for palette items. Here you put it into the context for both board and palette. This should not be happening

Reply:
The show preview is for items that are inserting from the palette to the board. The acquired item is not rendered by the palette as it must be a part of the board's grid. That's why showPreview is attached to the item. However, in the item the prop can be called differently.

Reply:
If you struggle to find a solution yourself, we can pair on finding a solution, but the current form is not mergeable

Reply:
I updated the naming - please check if that works better.

Alternatively, the following can work in theory:

```
function Palette() {
  const [dropState, setDropState] = useState<{ id: string, isExpanded: boolean; isAcquired: boolean }>(); 
  
  useDragSubscription("start", ({ draggableItem: { id } }) => {
    setDropState({ id, isExpanded: false, isAcquired: false });

    if (items.some((it) => it.id === id)) {
      setAnnouncement(i18nStrings.liveAnnouncementDndStarted);
    } else {
      setAnnouncement("");
    }
  });

  useDragSubscription("update", ({ draggableItem: { id }, dropTarget }) => {
    setDropState({ id, isExpanded: !!dropTarget, isAcquired: false });
  });

  useDragSubscription("submit", () => {
    setDropState(undefined);

    if (dropState && !dropState.isAcquired && items.some((it) => it.id === dropState.id)) {
      setAnnouncement(i18nStrings.liveAnnouncementDndDiscarded);
    }
  });

  useDragSubscription("discard", () => {
    setDropState(undefined);

    if (dropState && !dropState.dropState && items.some((it) => it.id === dropState?.id)) {
      setAnnouncement(i18nStrings.liveAnnouncementDndDiscarded);
    }
  });

  useDragSubscription("acquire", ({ draggableItem }) => {
    if (items.some((it) => it.id === draggableItem.id)) {
      setDropState({ id: draggableItem.id, isExpanded: false, isAcquired: true });
    }
  });
}

renderItem({ showPreview: item.id === dropState.id && (dropState.isExpanded || dropState.isAcquired) })
```

But there are challenges as we capture the palette children. It can work if children were not a callback () => ReactNode, but that will break the caching and damage the performance.